### PR TITLE
Dcr 65 order management

### PR DESCRIPTION
**Pull request by amrarick26**
Display the Name of the UserGroup that equals order.xp.CustomerNumber
Filter by UserGroup

### CODE REVIEW

File: src/app/orderManagement/orders/js/orders.config.js [hunk 1]
```diff
@@ -18,10 +18,35 @@ function OrdersConfig($stateProvider) {
...
                 OrderList: function(ocOrdersService, OrderCloud, Parameters) {
-                    return ocOrdersService.List(Parameters);
+                    return ocOrdersService.List(Parameters)
+                        .then(function(orders) {
+                            if(orders.Items.length) {
+                                return orders;
+                            } else {
+                                return null;
+                            }
+                        });
```
Comment (crhistianramirez) [inline src/app/orderManagement/orders/js/orders.config.js]: Why did you change this?

File: src/app/orderManagement/orders/js/orders.config.js [hunk 2]
```diff
@@ -18,10 +18,35 @@ function OrdersConfig($stateProvider) {
...
                 },
                 BuyerCompanies: function(OrderCloud) {
                     return OrderCloud.Buyers.List(null, 1, 100);
+                },
+                UserGroupsList: function($q, OrderCloud, Parameters, BuyerCompanies) {
+                    var queue = [];
+                    _.each(BuyerCompanies.Items, function(buyer) {
+                        queue.push(function(){
+                            var defer = $q.defer();
```
Comment (crhistianramirez) [inline src/app/orderManagement/orders/js/orders.config.js]: don't use $q.defer();

The reason we don't want to use them is that it wraps something that is already a promise, in another promise. If there are any errors, they aren't correctly propagated to the error handler and cause hard to debug situations.

**Review by crhistianramirez — CHANGES_REQUESTED** [review]

File: src/app/orderManagement/orders/js/orders.config.js (hunk 1 above)
Comment (amrarick26) [reply src/app/orderManagement/orders/js/orders.config.js]: I'll check this again, but before it was just freezing up if there were no items being returned.

**Review by crhistianramirez — APPROVED** [review]
